water/lib/CAPI/Dialects.cpp

```diff
@@ -41,6 +43,28 @@ MlirTypeID mlirWaveSymbolAttrGetTypeID() {
   return wrap(mlir::TypeID::get<wave::WaveSymbolAttr>());
 }
 
+//===---------------------------------------------------------------------===//
+// WaveIndexSymbolAttr
+//===---------------------------------------------------------------------===//
+
+bool mlirAttributeIsAWaveIndexSymbolAttr(MlirAttribute attr) {
+  return llvm::isa<wave::WaveIndexSymbolAttr>(unwrap(attr));
+}
+
+MlirAttribute mlirWaveIndexSymbolAttrGet(MlirContext mlirCtx, uint32_t value) {
```

Comment:
Nit: is `unit32_t` the type actually used to back the enum?

Reply:
yes. we are using `I32EnumAttr`